rest-client/src/test/java/com/fwmotion/threescale/cms/ThreescaleCmsClientImplUnitTest.java

```diff
@@ -796,10 +796,79 @@ void publish() {
     }
 
     @Test
-    void delete() {
+    void delete_File() throws Exception{
+        // Given a CmsFile object with an ID already
+        CmsFile newFile = new CmsFile();
+        newFile.setId(16);
+        newFile.setPath("/file.jpg");
+        newFile.setSectionId(30);
+        newFile.setTags(Set.of("a", "b", "c"));
+        newFile.setDownloadable(true);
+
+        // When the interface code is called
+        threescaleCmsClient.delete(newFile);
+
+        // Then only the file should have been deleted
+        then(filesApi).should(only()).deleteFile(
+            eq(newFile.getId()));
+        then(sectionsApi).shouldHaveNoInteractions();
+        then(templatesApi).shouldHaveNoInteractions();
     }
 
     @Test
-    void testDelete() {
+    void delete_FileById() throws Exception{
+        // Given a CmsFile object with an ID already
+        CmsFile newFile = new CmsFile();
+        newFile.setId(16);
+        newFile.setPath("/file.jpg");
+        newFile.setSectionId(30);
+        newFile.setTags(Set.of("a", "b", "c"));
+        newFile.setDownloadable(true);
+
+        // When the interface code is called
+        threescaleCmsClient.delete(newFile.getType(), newFile.getId());
```

Comment:
No need to build a whole `CmsFile` object just to use 2 properties from it. You can simplify like this: `threescaleCmsClient.delete(ThreescaleObjectType.FILE, 16);`